R/latent_model.R

```diff
@@ -89,12 +89,76 @@ epidist_family_model.epidist_latent_model <- function(
     ub = c(NA, as.numeric(lapply(family$other_bounds, "[[", "ub"))),
     type = family$type,
     vars = c("pwindow", "swindow", "vreal1"),
-    loop = FALSE
+    loop = FALSE,
+    log_lik = epidist_gen_log_lik_latent(family),
+    posterior_predict = epidist_gen_posterior_predict(family),
+    posterior_epred = epidist_gen_posterior_epred(family)
   )
   custom_family$reparm <- family$reparm
   return(custom_family)
 }
 
+#' Create a function to calculate the pointwise log likelihood of the latent
+#' model
+#'
+#' This function creates a log likelihood function that accounts for the latent
+#' variables in the model, including primary and secondary event windows and
+#' their overlap. The returned function calculates the log likelihood for a
+#' single observation by augmenting the data with the latent variables and
+#' using the underlying brms log likelihood function.
+#'
+#' @seealso [brms::log_lik()] for details on the brms log likelihood interface.
+#'
+#' @inheritParams epidist_family_model
+#'
+#' @return A function that calculates the log likelihood for a single
+#' observation. The prep object must have the following variables:
+#' * `vreal1`: relative observation time
+#' * `vreal2`: primary event window
+#' * `vreal3`: secondary event window
+#'
+#' @family latent_model
+#' @autoglobal
+epidist_gen_log_lik_latent <- function(family) {
+  # Get internal brms log_lik function
+  log_lik_brms <- .get_brms_fn("log_lik", family)
+
+  .log_lik <- function(i, prep) {
```

Comment:
So I guess it's not possible to replace this part with calls to `primarycensored` functions?

Reply:
no because this isn't the primarycensored model / likelihood